### PR TITLE
spago: 0.20.0 -> 0.20.1

### PR DESCRIPTION
**Pull request by cdepillabout**
###### Motivation for this change

There was a new version of `spago` released: https://github.com/purescript/spago/releases/tag/0.20.1

There were a few small changes with documentation-related stuff: https://github.com/purescript/spago/pull/775

cc @f-f

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cdepillabout** [comment]
@GrahamcOfBorg build spago

File: pkgs/development/tools/purescript/spago/spago.nix [hunk 1]
```diff
@@ -1,39 +1,40 @@
...
-    prettyprinter process retry rio rio-orphans safe semver-range stm
-    stringsearch tar template-haskell temporary text time transformers
-    turtle unliftio unordered-containers utf8-string vector versions
-    with-utf8 zlib
+    file-embed filepath foldl fsnotify generic-lens Glob http-client
+    http-conduit http-types lens-family-core megaparsec mtl network-uri
+    open-browser optparse-applicative prettyprinter process retry rio
+    rio-orphans safe semver-range stm stringsearch tar template-haskell
+    temporary text time transformers turtle unliftio
+    unordered-containers utf8-string versions with-utf8 zlib
```
Comment (SuperSandro2000) [inline pkgs/development/tools/purescript/spago/spago.nix:32]: Such reflows make reviewing a bit harder.

Comment (cdepillabout) [reply pkgs/development/tools/purescript/spago/spago.nix:32]: Yeah, sorry about this.

This should just be a generated file, but it is being generated with an older version of `cabal2nix`, so it doesn't have the fix for `stdenv.lib`.  I had to do that by hand, which made this a little messy.

The generation process for this file could be cleaned up a bit, but it is out of scope for this simple bump PR.

**Comment by cdepillabout** [comment]
This builds on ofborg, and it looks like tests pass locally, so I am going to go ahead and merge it in.

File: pkgs/development/tools/purescript/spago/spago.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/purescript/spago/spago.nix:32]: I didn't know cabal2nix can do this. I thought this is done by hand.

**Pull request merged**